profilehooks.py

```diff
@@ -750,10 +771,28 @@ def __call__(self, *args, **kw):
                 funcname = fn.__name__
                 filename = fn.__code__.co_filename
                 lineno = fn.__code__.co_firstlineno
-                sys.stderr.write("\n  %s (%s:%s):\n    %.3f seconds\n\n" % (
-                    funcname, filename, lineno, duration
-                ))
-                sys.stderr.flush()
+                args_str = ', '.join(str(item) for item in args)
+                kw_str = ', '.join(['%s=%s' % (key, value) for key, value in kw.items()])
+                arguments_str = args_str
+                if kw_str:
+                    if args_str:
+                        arguments_str += ', '
+                    arguments_str += kw_str
+                message = "%s:line %s:%s(%s):%.3f seconds" % (
+                    filename,
+                    lineno,
+                    funcname,
+                    arguments_str,
+                    duration
+                )
```

Comment:
Please move this logic inside the `if logger`, since you're not using the computed values in the `else` branch.

I don't like this sudden formatting of function arguments, as it opens a can of worms:
- why str() and not repr()?
- what if str() raises an exception?
- what if the argument is a list with a zillion items?

Let's make it log just the function name, filename, line number and duration, like in the log-less case.  Printing argument values may be a useful feature, but let's consider it separately from this PR.